libs/isograph-react/src/react/useLazyReference.ts

```diff
@@ -3,12 +3,44 @@ import {
   ExtractParameters,
 } from '../core/FragmentReference';
 import { useIsographEnvironment } from './IsographEnvironmentProvider';
-import { IsographEntrypoint } from '../core/entrypoint';
+import {
+  IsographEntrypoint,
+  type NormalizationAst,
+  type NormalizationAstLoader,
+} from '../core/entrypoint';
 import { getOrCreateCacheForArtifact } from '../core/cache';
 import { useLazyDisposableState } from '@isograph/react-disposable-state';
 import { logMessage } from '../core/logging';
-import { FetchOptions } from '../core/check';
+import { FetchOptions, type RequiredFetchOptions } from '../core/check';
 
+export function useLazyReference<
+  TReadFromStore extends { parameters: object; data: object },
+  TClientFieldValue,
+>(
+  entrypoint: IsographEntrypoint<
+    TReadFromStore,
+    TClientFieldValue,
+    NormalizationAstLoader
+  >,
+  variables: ExtractParameters<TReadFromStore>,
+  fetchOptions: RequiredFetchOptions<TClientFieldValue>,
+): {
+  fragmentReference: FragmentReference<TReadFromStore, TClientFieldValue>;
+};
+export function useLazyReference<
+  TReadFromStore extends { parameters: object; data: object },
+  TClientFieldValue,
+>(
+  entrypoint: IsographEntrypoint<
+    TReadFromStore,
+    TClientFieldValue,
+    NormalizationAst
+  >,
+  variables: ExtractParameters<TReadFromStore>,
+  fetchOptions?: FetchOptions<TClientFieldValue>,
+): {
+  fragmentReference: FragmentReference<TReadFromStore, TClientFieldValue>;
+};
 export function useLazyReference<
```

Comment:
Overloading makes an ugly error message if ast is lazy loaded then for `useLazyReference(entrypoint, {})` the error will be `NormalizationAstLoader` is not assignable to `NormalizationAst`, instead of missing argument error.

Reply:
🤔